catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -144,13 +145,24 @@ public UpdateRequest process(UpdateRequest input)
             }
         }
 
+        Attribute validationErrors = metacard.getAttribute(Validation.VALIDATION_ERRORS);
+        Attribute validationWarnings = metacard.getAttribute(Validation.VALIDATION_WARNINGS);
+
+        if (validationErrors != null) {
+            errors.addAll(validationErrors.getValues());
+            valid = INVALID_TAG;
+        }
+
+        if (validationWarnings != null) {
+            warnings.addAll(validationWarnings.getValues());
+            valid = INVALID_TAG;
+        }
+
         tags.add(valid);
```

Comment:
❓ Instead of clearing and setting the `valid` flag everywhere in the method, should we just have one check here to see if there are any errors or warnings and update the tags as needed?